packages/server/graphql/public/typeDefs/addApprovedOrganizationDomains.graphql

```diff
@@ -0,0 +1,35 @@
+extend type Mutation {
+  """
+  Restrict accepting team invites to a list of approved domains
+  """
+  addApprovedOrganizationDomains(
+    """
+    The organization ID
+    """
+    orgId: ID!
+
+    """
+    A list of domains or email addressed allowed to join the organization
+    """
+    emailDomains: [String!]!
+  ): AddApprovedOrganizationDomainsPayload!
+}
+
+union AddApprovedOrganizationDomainsPayload = ErrorPayload | AddApprovedOrganizationDomainsSuccess
+
+type AddApprovedOrganizationDomainsSuccess {
+  orgId: ID!
+
+  """
+  Organization with the updated approvals
+  """
+  organization: Organization!
+}
+
+extend type Organization {
```

Comment:
by extending here, we get type safety in the resolver & don't have to touch the legacy /graphql/types/Organization.ts